### PR TITLE
Fix panic in Pattern.Matches with zero StarIndex

### PR DESCRIPTION
**Pull request by abelcha**
This PR fixes a panic that occurs in the Pattern.Matches method when StarIndex is 0 and Text is empty or too short. The panic was caused by attempting to slice Text with invalid bounds [1:0].

The fix adds special handling for edge cases:
- Empty Text
- StarIndex at the beginning of Text
- StarIndex at or beyond the end of Text

Added tests to verify the fix:
- Unit tests in internal/core/pattern_test.go
- Integration test in internal/module/circular_reference_test.go
- Compiler test case in testdata/tests/cases/compiler/circularModuleReference.ts

Fixes the issue with circular module references in eslint.config.js files.

### CODE REVIEW

**Comment by abelcha** [comment]
@microsoft-github-policy-service agree

**Comment by abelcha** [comment]
> @abelcha please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
>  > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

File: internal/core/pattern_test.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+func TestCircularModuleReference(t *testing.T) {
+	// This test simulates the scenario with the eslint.config.js file
+	// that contains: export { default } from '../eslint.config.js'
+	
+	// Create a pattern that might be used in module resolution
+	// The key issue is when Text is empty but StarIndex is 0
+	pattern := Pattern{
+		Text:      "",
+		StarIndex: 0,
+	}
```
Comment (DanielRosenwasser) [inline internal/core/pattern_test.go:49]: This should never happen - the empty string is an exact match (`StarIndex == -1`).

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]
I think this was attempting to fix #870 but #875 already did that. Any details on what else this might have fixed?

**Comment by jakebailey** [comment]
I think #875 fixed this
